book/src/SUMMARY.md

```diff
@@ -5,7 +5,112 @@
 
 # User Guide
 
+## Multi-threading
+
+Riot-rs implements a scheduler based on fixed priorities and preemption.
+Within the same priority level, threads are scheduled cooperatively.
+
+Threading can be enabled with the `threading` feature.
+If the feature is enabled, at least one thread **must** be defined.  
+Threads should be implemented using the `riot_rs_macros::thread` proc macro, which takes care of calling the necessary initialization methods and linking the thread function element it into the binary.
+A `ThreadId` between 0 and `riot_rs_threads::THREADS_NUMOF` is assigned to each thread in the order in which the threads are declared (*TODO: is this a defined behavior? Would ease e.g. setting thread\_flags for other threads because the ids are known at compile time, but is it really always ensured?*).
+Optionally, the stacksize and a priority between 1 and `riot_rs_threads::SCHED_PRIO_LEVELS` can be configured. Per default, the stack size is 2048 bytes and priority is 1.
+
+The `riot_rs_threading` crate supports basic synchronization primitives:
+- a synchronous (blocking) channel for sending data between threads (the data must implement `Copy`)
+- a lock that can be used to implement mutually exclusive access to a resource
+- a thread-flags implementation that enables threads to wait for specific flags
+  (*TODO: should the thread\_flags even be mentioned here/ are they meant to be used by users?*)
+
+### Example
+
+```rs
```

Comment:
As asked elsewhere: this would be useful as an example. maybe just copy `examples/threading` to `examples/threading_channel`, put his code there, then use here via mdbook include (with anchor)?

(We can postpone that to a follow up)

Reply:
Implemented in #223.